app/views/admin/unattached_promo_registrations/index.html.slim

```diff
@@ -1,110 +1,113 @@
-              .flex-one.align-self-flex-center
-                . style="font-weight: bold" = label_tag = "Event types"
-                .
-                  = check_box_tag "event_types[]", PromoRegistration::RETRIEVALS
-                  = label_tag = event_type_column_header(PromoRegistration::RETRIEVALS) + " "
-                .
-                  = check_box_tag "event_types[]", PromoRegistration::FIRST_RUNS
-                  = label_tag = event_type_column_header(PromoRegistration::FIRST_RUNS) + " "
-                .
-                  = check_box_tag "event_types[]", PromoRegistration::FINALIZED
-                  = label_tag = event_type_column_header(PromoRegistration::FINALIZED) + " "
-              .flex-one.align-self-flex-end
-                = submit_tag "Download", id: "download-referral-reports", class: "btn btn-primary", data: { disable_with: false }
+            .flex
+              .flex-one = select_tag :referral_code_status, options_for_select([["Active", "active"], ["Paused", "paused"]])
+              = hidden_field_tag :filter, params[:filter]
+              = submit_tag "Update", id: "update-referral-code-statuses", class: "btn btn-primary"
+      .flex
+        .panel
+          .panel--sub-panel-header = "Report generation"
+          hr
+          .flex
+            .flex-two.align-self-flex-center
+              . = label_tag "Report period start"
+              . = date_select :referral_code_report_period, :start
+            .flex-two.align-self-flex-center
+              . = label_tag "Report period end"
+              . = date_select :referral_code_report_period, :end
+            .flex-two.align-self-flex-center
+              . = label_tag "Reporting interval"
+              . = select_tag :reporting_interval, options_for_select([["Daily",PromoRegistration::DAILY], ["Weekly",PromoRegistration::WEEKLY], ["Monthly",PromoRegistration::MONTHLY], ["Running total", PromoRegistration::RUNNING_TOTAL]])
+            .flex-one.align-self-flex-center
+              . style="font-weight: bold" = label_tag = "Event types"
+              .
+                = check_box_tag "event_types[]", PromoRegistration::RETRIEVALS
+                = label_tag = " " + event_type_column_header(PromoRegistration::RETRIEVALS)
+              .
+                = check_box_tag "event_types[]", PromoRegistration::FIRST_RUNS
+                = label_tag = " " + event_type_column_header(PromoRegistration::FIRST_RUNS)
+              .
+                = check_box_tag "event_types[]", PromoRegistration::FINALIZED
+                = label_tag = " " + event_type_column_header(PromoRegistration::FINALIZED)
+            .flex-one.align-self-flex-end
+              = submit_tag "Download", id: "download-referral-reports", class: "btn btn-primary", data: { disable_with: false }
```

Comment:
https://api.rubyonrails.org/v5.1/classes/ActionView/Helpers/FormTagHelper.html

Preferably we'd replace `disable_with: 'Please wait'`

Reply:
Actually nvm, it's fine